<launch>

	<param name="/use_sim_time" value="true"/>

	<include file="$(find localization_launch)/launch/localization_launch.launch" />

	<!--  ****** Maps *****  -->
	<node name="map_server" pkg="map_server" type="map_server" args="$(find move_base_ssu)/maps/back_of_ssara.yaml" output="screen">
		<param name="frame_id" value="map"/>
	</node>>

	<!-- ****** Move_Base ****** -->
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
  	  	<rosparam file="$(find move_base_ssu)/cfg/costmap_common_params.yaml" command="load" ns="global_costmap" />
  	 	<rosparam file="$(find move_base_ssu)/cfg/costmap_common_params.yaml" command="load" ns="local_costmap" />
  		<rosparam file="$(find move_base_ssu)/cfg/local_costmap_params.yaml" command="load" />
  		<rosparam file="$(find move_base_ssu)/cfg/global_costmap_params.yaml" command="load" />
  		<rosparam file="$(find move_base_ssu)/cfg/teb_local_planner_params.yaml" command="load" />

		<param name="base_global_planner" value="global_planner/GlobalPlanner" />
		<param name="planner_frequency" value="1.0" />
		<param name="planner_patience" value="5.0" />

		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
		<param name="controller_frequency" value="5.0" />
		<param name="controller_patience" value="15.0" />

    	<param name="clearing_rotation_allowed" value="false" /> <!-- Our carlike robot is not able to rotate in place -->
	</node>


	<!--  ****** ACML *****  -->
	<node pkg="amcl" type="amcl" name="amcl" output="screen">
		<rosparam file="$(find move_base_ssu)/cfg/amcl_params.yaml" command="load" />

	    <param name="initial_pose_x"            value="0"/>
		<param name="initial_pose_y"            value="0"/>
		<param name="initial_pose_a"            value="90"/>
	</node>
</launch>